### PR TITLE
Update types

### LINKED ISSUE #39: Look into when to use React node, React Element or JSX Element as types
#### Describe the desired behavior
Need a unified rule for when to use React node, React Element or JSX Element as return types & propTypes

#### Describe the current behavior
Nearly everything is JSX Element

https://github.com/pxblue/react-native-component-library/issues/15

### PR DESCRIPTION
**Pull request by joebochill**
Fixes #39.

Changes proposed in this Pull Request:
- Updates some type definitions and docs based on better understanding of ReactElement, ReactNode, and JSX.Element

### CODE REVIEW

File: docs/Hero.md [hunk 1]
```diff
@@ -44,13 +44,13 @@ import { Hero } from '@pxblue/react-components';
 | ------------------- | --------------------------------------- | ----------------------------- | -------- | ------------- |
 | classes             | Style overrides                         | `StyleRules`                  | no       |               |
 | fontSize            | The text size for the value line        | `'normal'` \| `'small'`       | no       | 'normal'      |
-| icon                | The primary icon                        | `React.Component` \| `string` | yes      |               |
+| icon                | The primary icon                        | `HeroClasses`                 | yes      |               |
```
Comment (huayunh) [inline docs/Hero.md:47]: Typo? icon is type HeroClasses?

Comment (joebochill) [reply docs/Hero.md]: Good catch, fix coming

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 ## V4.0.2
 
-- Change title and description types from `string` to `ReactNode`.
+- Updated several prop types to `ReactNode` to support wider range of input values.
```
Comment (daileytj) [inline CHANGELOG.md:5]: I updated this to be "Change title and description types from `string` to `ReactNode` for `<EmptyState>`." In the PR for the info list item third line of text. Which one do you want to go with?

Comment (daileytj) [reply CHANGELOG.md:5]: Nevermind, that was a pre-emptive comment, I didn't look at the rest of your PR...Your changelog update makes sense so just ignore that last comment.

**Pull request merged**